src/dhtproto/client/UsageExamples.d

```diff
@@ -878,6 +878,119 @@ unittest
     }
 }
 
+/// Example of neo Update request usage
+unittest
+{
+    class UpdateExample : ExampleApp
+    {
+        import ocean.core.array.Mutation : copy;
+
+        override protected void example ( )
+        {
+            // Assign a neo Update request. Note that the channel is copied
+            // inside the client -- the user does not need to maintain it after
+            // calling this method.
+            this.dht.neo.update("channel", 0x1234567812345678, &this.updateNotifier);
+        }
+
+        // Notifier which is called when something of interest happens to
+        // the Update request. See dhtproto.client.request.Update for
+        // details of the parameters of the notifier.
+        private void updateNotifier ( DhtClient.Neo.Update.Notification info,
+            Const!(DhtClient.Neo.Update.Args) args )
+        {
+            formatNotification(info, this.msg_buf);
+
+            with ( info.Active ) final switch ( info.active )
+            {
+                case received:
+                    // Set updated value. In this simple example, we simply
+                    // copy the original value, without change.
+                    auto received_record = info.received.value;
+                    (*info.received.updated_value).copy(info.received.value);
+                    goto case;
```

Comment:
Can you explain what this `goto` will do, and why?

Reply:
Fall-through to the next case (required by D2).

Reply:
Why? So that `log.trace` is called for this notification.

Reply:
Ah, gotcha.  I don't think I'd ever come across that syntax before, good to know.

Reply:
Yeah, it's kinda weird. Implicit fall-through is disallowed in D2.